### ISSUE TITLE
port_map doesn't work

### ISSUE THREAD

**Issue by ChrisMcKenzie** [hashicorp/nomad#472]
I maybe misunderstanding but when I configure the `port_map` section of `task.config` it should take the ports given in that map and use them as the port the port that I am listening on in my container and map it to the port that was randomly selected on the host. If this correct it does not appear to be doing that it expose the port from the map but it is not mapped to the host ports 

docker reports the following ports 
![](http://share.chrismckenzie.io/1._tmux_2015-11-19_14-53-37.png)

__Note:__ this is happening with even the example.nomad generated by running nomad init

**Comment by cbednarski** [comment]
Thanks for the report. Can you show me the task config you're using?

There was a similar PR merged this morning to fix the wrong port being exposed but it looked like `port_map` was working correctly here. https://github.com/hashicorp/nomad/pull/466

**Comment by ChrisMcKenzie** [comment]
I think that #466 is exactly what I am experiencing. 

for what its worth I have been trying a few different task but I was unable to even get this example.nomad to work I get the same looking port map as in #466
```
task "redis" {
  # Use Docker to run the task.
  driver = "docker"

  # Configure Docker driver with the image
  config {
    image = "redis:latest"
    port_map {
      db = 6379
    }
  }

  service {
    # name = redis
    tags = ["global", "cache"]
    port = "db"
    check {
      name = "alive"
      type = "tcp"
      interval = "10s"
      timeout = "2s"
    }
  }

  # We must specify the resources required for
  # this task to ensure it runs on a machine with
  # enough capacity.
  resources {
    cpu = 500 # 500 Mhz
    memory = 256 # 256MB
    network {
      mbits = 10
      port "db" {
      }
    }
  }
}
```

**Comment by diptanu** [comment]
@ChrisMcKenzie This is already fixed in master, we will release a binary of 0.2.1 this Friday, in the meantime let us know if you see any other issues.

**Comment by ChrisMcKenzie** [comment]
Thanks so much!
